pyqtgraph/graphicsItems/ImageItem.py

```diff
@@ -323,6 +324,24 @@ def setAutoDownsample(self, active: bool=True):
         self._renderRequired = True
         self.update()
 
+    def setNanPolicy(self, nanPolicy: str):
+        """
+        Control how NaN values are handled during downsampling for this ImageItem.
+
+        Parameters
+        ----------
+        nanPolicy : str
+            Must be one of ['ignore`, 'propagate']. If 'nanPolicy' is 'ignore', NaNs
+            are automatically ignored during downsampling, at the expense of
+            performance. If 'nanPolicy' is 'propagate', NaNs are kept during
+            downsampling. ImageItem is instantiated with nanPolicy set to 'ignore'.
+        """
+        assert nanPolicy in ['propagate', 'ignore'], (f"{nanPolicy=} must be one "
+                                                      f"of ['propagate', 'ignore']")
+        self.nanPolicy = nanPolicy
```

Comment:
Since this is catching user errors, it should probably raise a ValueError (instead of using assert).

Reply:
Agreed if nanPolicy is not in the above, we can raise a `ValueError`, let's not have an `assert` check here.

Reply:
Also rename the attribute to `._nanPolicy`, see comment below about getter/setter.